From 7aaa4d23125c981b55868ec43a44c22aeb67600e Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Mon, 6 May 2013 16:39:08 +0700
Subject: Dalvik: BE error framework should restore the global flags

BZ: 105858

Backend error framework does not restore the global flags if
compilation of some trace was rejected.
The patch adds an invocation of restore method before bail out.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I4b2d2e50d011f0cb0e36a3cc2a797a53651a2ccf
Orig-MCG-Change-Id: I6d02e250816c1fec5655032a5778c40cf35fd524
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/CodegenInterface.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/codegen/x86/CodegenInterface.cpp b/vm/compiler/codegen/x86/CodegenInterface.cpp
index 50c9dce..05ee268 100644
--- a/vm/compiler/codegen/x86/CodegenInterface.cpp
+++ b/vm/compiler/codegen/x86/CodegenInterface.cpp
@@ -2392,6 +2392,8 @@ void dvmCompilerMIR2LIR(CompilationUnit *cUnit, JitTranslationInfo *info) {
             ALOGI("Too many retries while compiling trace  %s%s, offset %d", cUnit->method->clazz->descriptor,
                 cUnit->method->name, cUnit->traceDesc->trace[0].info.frag.startOffset);
             ALOGI("Rejecting Trace");
+            //Restore the compilation state
+            dvmRestoreCompilationState (compilationState);
             return;
         }
 
-- 
1.7.4.1

